appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy

```diff
@@ -1544,25 +1544,28 @@ class ActivityTest extends AbstractAppServiceIntegrationTest
 		appService.deleteUser(richardDefault)
 	}
 
-	def 'Richard\'s \'non-no-go\' network activity is measured on iOS only'(def operatingSystem, def expectedValue)
+	def 'Richard\'s network activity is measured properly on different OSes and different budget goals'(def operatingSystem, def budget, def expectedValue)
 	{
 		given:
 		User richard = addRichard(false, operatingSystem)
 		setCreationTimeOfMandatoryGoalsToNow(richard)
-		addTimeZoneGoal(richard, SOCIAL_ACT_CAT_URL, ["01:00-12:00"], "W-1 Mon 02:18")
+		addBudgetGoal(richard, SOCIAL_ACT_CAT_URL, budget, "W-1 Mon 02:18")
 		richard = appService.reloadUser(richard, CommonAssertions.&assertUserGetResponseDetailsWithPrivateDataIgnoreDefaultDevice)
-		Goal timeZoneGoalSocialRichard = richard.findActiveGoal(SOCIAL_ACT_CAT_URL)
+		BudgetGoal budgetGoalSocialRichard = richard.findActiveGoal(SOCIAL_ACT_CAT_URL)
 
 		// Activities on default device
 		reportNetworkActivity(richard, ["social"], "http://www.facebook.com", "W-1 Mon 03:20")
+		reportNetworkActivity(richard, ["social"], "http://www.facebook.com", "W-1 Mon 03:24")
 
+		def expectedGoalAccomplished = budget > 0
+		def expectedMinutesBeyondGoal = expectedGoalAccomplished ? 0 : expectedValue.size() == 0 ? 0 :  expectedValue.entrySet().iterator().next().value
```

Comment:
I would prefer a simple test parameter instead of this computation

Reply:
That would lead to duplication of information, which I'm not in favor of.
How about rewriting the expression? Is this better?
```
def expectedMinutesBeyondGoal = expectedGoalAccomplished || expectedValue.size() == 0 ? 0 :  expectedValue.entrySet().iterator().next().value
```

Reply:
I still prefer the duplication of information in tests above using code to calculate information. But if you prefer it this way, we keep it this way.

Reply:
OK. I forgot to push the simplification, but now I've done it. A nested ternary operator is not a nice thing.